Moesif.Middleware/NetCore/MoesifMiddlewareNetCore.cs

```diff
@@ -950,15 +775,9 @@ private String getUserId(HttpContext httpContext, EventRequestModel request)
         private async Task LogEventAsync(EventModel eventModel)
         {
 #if MOESIF_INSTRUMENT
-            Stopwatch stopwatch = new Stopwatch();
-            stopwatch.Start();
-
-            long getMaskEventTime = 0;
-            long createRequestMapTime = 0;
-            long samplingPercentageTime = 0;
-            long randomPercentageTime = 0;
-            long computeWeightTime = 0;
-            long createEventAsyncTime = 0;
+            var logStage = false;
+            var perfMetrics = new PerformanceMetrics("LogEventAsync", logStage);
+            perfMetrics.Start("getMaskEvent");
 #endif
 
             // Get Mask Event : REVIEW can it be created early?
```

Comment:
`MaskEvent` functionality initialization should also move out - will move it out before publishing.

Reply:
There is a corresponding function in .NetFramework too.